qiskit/transpiler/passes/scheduling/calibration_creators.py

```diff
@@ -0,0 +1,222 @@
+        """
+        return isinstance(node_op, RZXGate)
+
+    @staticmethod
+    def rescale_cr_inst(instruction: Play, theta: float, sample_mult: int = 16) -> Play:
+        """
+        Args:
+            instruction: The instruction from which to create a new shortened or lengthened pulse.
+            theta: desired angle, pi/2 is assumed to be the angle that the pulse in the given
+                play instruction implements.
+            sample_mult: All pulses must be a multiple of sample_mult.
+
+        Returns:
+            Play: The play instruction with the stretched compressed GaussianSquare pulse.
+
+        Raises:
+            QiskitError: if the pulses are not GaussianSquare.
+        """
+        pulse_ = instruction.pulse
+        if isinstance(pulse_, GaussianSquare):
+            amp = pulse_.amp
+            width = pulse_.width
+            sigma = pulse_.sigma
+            n_sigmas = (pulse_.duration - width) / sigma
+
+            # The error function is used because the Gaussian may have chopped tails.
+            gaussian_area = abs(amp) * sigma * np.sqrt(2 * np.pi) * math.erf(n_sigmas)
+            area = gaussian_area + abs(amp) * width
+
+            target_area = abs(theta) / (np.pi / 2.) * area
+            sign = theta / abs(theta)
+
+            if target_area > gaussian_area:
+                width = (target_area - gaussian_area) / abs(amp)
+                duration = math.ceil((width + n_sigmas * sigma) / sample_mult) * sample_mult
+                return Play(GaussianSquare(amp=sign*amp, width=width, sigma=sigma,
+                                           duration=duration), channel=instruction.channel)
+            else:
+                amp_scale = sign * target_area / gaussian_area
+                duration = math.ceil(n_sigmas * sigma / sample_mult) * sample_mult
+                return Play(
+                    GaussianSquare(amp=amp * amp_scale, width=0, sigma=sigma, duration=duration),
+                    channel=instruction.channel)
+        else:
+            raise QiskitError('RZXCalibrationBuilder only stretches/compresses GaussianSquare.')
+
+    def get_calibration(self, params: List, qubits: List) -> Schedule:
+        """
+        Args:
+            params: Parameters of the RZXGate(theta). I.e. params[0] is theta.
+            qubits: List of qubits for which to get the schedules. The first qubit is
+                the control and the second is the target.
+
+        Returns:
+            schedule: The calibration schedule for the RZXGate(theta).
+
+        Raises:
+            QiskitError: if the the control and target qubits cannot be identified.
+        """
+        theta = params[0]
+        q1, q2 = qubits[0], qubits[1]
+        cx_sched = self._inst_map.get('cx', qubits=(q1, q2))
+        rzx_theta = Schedule(name='rzx(%.3f)' % theta)
+
+        if theta == 0.0:
+            return rzx_theta
+
+        crs, comp_tones, shift_phases = [], [], []
```

Comment:
the way these are built seems pretty tied to the current echo-cross-resonance with rotary pulses, right? If a different flavor of CX is exposed can this handle it?

Reply:
> the way these are built seems pretty tied to the current echo-cross-resonance with rotary pulses, right?

Yes. Since the `RZXCalibrationBuilder` is creating schedules from the calibrated CX schedules of the backend it must be tied to the way the CX is built. This is in fact very important because by leveraging the calibrated CX gates we can get very good ZX(theta) for arbitrary theta without doing any extra calibration.

> If a different flavor of CX is exposed can this handle it?

No. If someone has a completely different way of building the CX then a new subclass of `CalibrationBuilder` will need to be written. However, some small extensions to `RZXCalibrationBuilder` could be made such as dealing with the presence and absence of rotary pulses and dealing with the presence and absence of echos.

Reply:
In d69dce8 I modified the code so that it can handle CX schedules with and without rotary tones.

Reply:
Should the class docstring include the assumptions being made about the CX implementation?

Reply:
Good point. I improved the docstring in c589104 and 47b63d6 which now reads:
```
    Creates calibrations for RZXGate(theta) by stretching and compressing
    Gaussian square pulses in the CX gate. This is done by retrieving (for a given pair of
    qubits) the CX schedule in the instruction schedule map of the backend defaults.
    The CX schdule must be an echoed cross-resonance gate optionally with rotary tones.
    The cross-resonance drive tones and rotary pulses must be Gaussian square pulses.
    The width of the Gaussian square pulse is adjusted so as to match the desired rotation angle.
    If the rotation angle is small such that the width disapears then the amplitude of the
    zero width Gaussian square pulse (i.e. a Gaussian) is reduced to reach the target rotation
    angle. Additional details can be found in https://arxiv.org/abs/2012.11660.
```